content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,170 @@
+---
+title: 'Prisma Client extensions'
+metaTitle: 'Prisma Client extensions (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+Use Prisma Client extensions to add functionality to the models in your schema, results objects, queries, or Prisma Client itself.
+
+There are four types of extension. You can use each type to customize a different aspect of Prisma Client.
+
+- `model` extensions: [add custom methods or fields to your models](/concepts/components/prisma-client/client-extensions/client-extensions-model)
+- `client` extensions: [add client-level methods to Prisma Client](/concepts/components/prisma-client/client-extensions/client-extensions-client)
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/client-extensions-result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/client-extensions-query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To create Prisma Client extensions, use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods), and the appropriate extension type (`model`, `client`, `query` or `result`).
```

Comment:
An example here maybe of any of them as illustration? Does not need to ahve actual extension logic but show the `$extends` usage and the extension type - and where the actual logic then would need to go.

Reply:
Extension name should probably also be mentioned and explained here (instead of on the subpages that explain specific extension types)

Reply:
I'll take a look at this introductory section again

Reply:
I've added a code example here to show the structure that users should use in their code. I think we need to explain  `name` on the other pages, too, because users might go straight there. Of course, we can't explain everything on every docs page, but we should make everything as self-contained as possible, and link out to prerequisite knowledge where we can't include it on a page.